modules/spring-config/src/main/java/org/mule/runtime/config/internal/AbstractSpringMuleContextServiceConfigurator.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.config.internal;
+
+import static org.mule.runtime.config.internal.InjectParamsFromContextServiceProxy.createInjectProviderParamsServiceProxy;
+import static org.mule.runtime.core.api.config.MuleProperties.OBJECT_STORE_MANAGER;
+
+import static java.lang.reflect.Proxy.getInvocationHandler;
+import static java.lang.reflect.Proxy.isProxyClass;
+
+import static org.springframework.beans.factory.support.BeanDefinitionBuilder.genericBeanDefinition;
+
+import org.mule.runtime.api.artifact.Registry;
+import org.mule.runtime.api.service.Service;
+import org.mule.runtime.config.internal.factories.ConstantFactoryBean;
+import org.mule.runtime.config.internal.factories.FixedTypeConstantFactoryBean;
+import org.mule.runtime.core.internal.config.CustomService;
+import org.mule.runtime.core.internal.config.CustomServiceRegistry;
+import org.mule.runtime.core.internal.util.TypeSupplier;
+import org.mule.runtime.module.service.internal.manager.LazyServiceProxy;
+
+import java.lang.reflect.InvocationHandler;
+import java.util.Optional;
+
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.beans.factory.support.BeanDefinitionBuilder;
+import org.springframework.beans.factory.support.BeanDefinitionRegistry;
+
+abstract class AbstractSpringMuleContextServiceConfigurator {
+
+  private final CustomServiceRegistry customServiceRegistry;
+  private final BeanDefinitionRegistry beanDefinitionRegistry;
+  private final Registry serviceLocator;
+
+  protected AbstractSpringMuleContextServiceConfigurator(CustomServiceRegistry customServiceRegistry,
+                                                         BeanDefinitionRegistry beanDefinitionRegistry,
+                                                         Registry serviceLocator) {
+    this.customServiceRegistry = customServiceRegistry;
+    this.beanDefinitionRegistry = beanDefinitionRegistry;
+    this.serviceLocator = serviceLocator;
+  }
+
+  protected void registerConstantBeanDefinition(String serviceId, Object impl) {
+    registerBeanDefinition(serviceId, getConstantObjectBeanDefinition(impl));
+  }
+
+  protected void registerBeanDefinition(String serviceId, BeanDefinition beanDefinition) {
```

Comment:
Add appropriate comments or try to separate this class in different methods/classes so that the cyclomatic complexity is reduced and why each condition should be met.